vllm/platforms/cuda.py

```diff
@@ -110,41 +121,63 @@ class CudaPlatform(Platform):
 
     @classmethod
     def get_device_capability(cls, device_id: int = 0) -> DeviceCapability:
-        physical_device_id = device_id_to_physical_device_id(device_id)
-        major, minor = get_physical_device_capability(physical_device_id)
+        with nvml_context() as nvml:
+            if nvml:
+                physical_device_id = device_id_to_physical_device_id(device_id)
+                major, minor = nvml.get_physical_device_capability(
+                    physical_device_id)
+            else:
+                major, minor = torch.cuda.get_device_capability(device_id)
```

Comment:
can you add another class like `NonNVMLContext` and implement it there? don't put too many `if-else` here.